dart/dynamics/Entity.h

```diff
@@ -63,11 +63,23 @@ class Shape;
 /// may have different policies about how/if their reference frame or name
 /// can be changed. Use the Detachable class to create an Entity whose reference
 /// Frame can be changed arbitrarily.
-class Entity
+class Entity : public common::Subscription
 {
 public:
   friend class Frame;
 
+  enum Notification {
```

Comment:
The compromise sounds good. I implemented Signal class in [grey/subscription_signal](https://github.com/dartsim/dart/commit/4bc93dcaf7e19349917ef55f8cb929516effcf72) branch. Event was renamed to Signal because its design is quiet close to Signal/Slot of Qt and Boost.Signals2.

While I was implementing Signal, I also considered Boost.Signals2 as an alternative since don't want to *reinvent the wheel*. Boost.Signals2 is mature library and thread safe. However, I found some comments that Boost.Signals2 shows low performance [[1]](https://testbit.eu/cpp11-signal-system-performance/) [[2]](http://www.cpgf.org/document/callback-benchmark.html). 

Here are some performance test with Publisher, Signal, and Boost.Signals2:
```
[ Number of listeners: 10 (slots), 50k calls]
Publisher                 : 0.00658989
Signal                    : 0.00805116
Boost.Signal              : 0.035176
Boost.Signal (dummy mutex): 0.0199561

[ Number of listeners: 100 (slots), 50k calls]
Publisher                 : 0.0706949
Signal                    : 0.0816331
Boost.Signal              : 0.274847
Boost.Signal (dummy mutex): 0.146624

[ Number of listeners: 1000 (slots), 50k calls]
Publisher                 : 0.822108
Signal                    : 0.843535
Boost.Signal              : 2.96526
Boost.Signal (dummy mutex): 1.62066
```
The unit is second, and the test code can be found [here](https://github.com/dartsim/dart/blob/js/signal_testing/unittests/testSubscriptions.cpp#L237).

Each test invoked 50k times for different number of slots. `Boost.Signal (dummy mutex)` were tested [without thread safety](http://theboostcpplibraries.com/boost.signals2-multithreading).

As for now, it seems Signal looks reasonable solution in efficiency and generalizability. On the other hand, I'm still open to other solutions if it's more proper for DART.